extension/jaegerremotesampling/extension_test.go

```diff
@@ -61,24 +66,57 @@ func TestStartAndShutdownRemote(t *testing.T) {
 	cfg := testConfig()
 	cfg.GRPCServerSettings.NetAddr.Endpoint = "127.0.0.1:0"
 	cfg.Source.Remote = &configgrpc.GRPCClientSettings{
-		Endpoint:     fmt.Sprintf("127.0.0.1:%d", lis.Addr().(*net.TCPAddr).Port),
+		Endpoint: fmt.Sprintf("127.0.0.1:%d", lis.Addr().(*net.TCPAddr).Port),
+		TLSSetting: configtls.TLSClientSetting{
+			Insecure: true, // test only
+		},
 		WaitForReady: true,
+		Headers: map[string]configopaque.String{
+			"testheadername": "testheadervalue",
+		},
 	}
 
 	// create the extension
 	e := newExtension(cfg, componenttest.NewNopTelemetrySettings())
 	require.NotNil(t, e)
 
-	// test
+	// start the server
 	assert.NoError(t, e.Start(context.Background(), componenttest.NewNopHost()))
+
+	// make a call
```

Comment:
Previously, this integration test was just starting/stopping – fortunately, it wasn't hard to make exercise the extension-hosted API (and in turn, to verify that the expected calls are made with the expected metadata)